bugwarrior/services/logseq.py

```diff
@@ -0,0 +1,301 @@
+import logging
+
+import requests
+import typing_extensions
+
+import re
+from datetime import datetime
+
+from bugwarrior import config
+from bugwarrior.services import IssueService, Issue, ServiceClient
+
+log = logging.getLogger(__name__)
+
+
+class LogseqConfig(config.ServiceConfig):
+    service: typing_extensions.Literal["logseq"]
+    host: str = "localhost"
+    port: int = 12315
+    token: str = ""
+    task_state: str = "DOING, TODO, NOW, LATER, IN-PROGRESS, WAIT, WAITING"
+    char_open_link: str = "【"
+    char_close_link: str = "】"
+    char_open_bracket: str = "〈"
+    char_close_bracket: str = "〉"
+    inline_links = True  # set to False to override bugwarrior config
+
+
+class LogseqClient(ServiceClient):
+    def __init__(self, host, port, token, filter):
+        self.host = host
+        self.port = port
+        self.token = token
+        self.filter = filter
+
+        self.headers = {
+            "Authorization": "Bearer " + self.token,
+            "content-type": "application/json; charset=utf-8",
+        }
+
+        self.project = None
+
+    def _datascript_query(self, query):
+        try:
+            response = requests.post(
+                f"http://{self.host}:{self.port}/api",
+                headers=self.headers,
+                json={"method": "logseq.DB.datascriptQuery", "args": [query]},
+            )
+            return self.json_response(response)
+        except requests.exceptions.ConnectionError as ce:
+            log.fatal("Unable to connect to Logseq HTTP APIs server. %s", ce)
+            exit()
+
+    def _get_current_graph(self):
+        try:
+            response = requests.post(
+                f"http://{self.host}:{self.port}/api",
+                headers=self.headers,
+                json={"method": "logseq.getCurrentGraph", "args": []},
+            )
+            return self.json_response(response)
+        except requests.exceptions.ConnectionError as ce:
+            log.fatal("Unable to connect to Logseq HTTP APIs server. %s", ce)
+            exit()
+
+    def get_graph_name(self):
+        graph = self._get_current_graph()
+        return graph["name"] if graph else None
+
+    def get_issues(self):
+        task_filter = self.filter.replace(" ", "").replace(",", '" "')
+        return self._datascript_query(
+            f"""
+            [:find (pull ?b [*])
+                :where [?b :block/marker ?marker]
+                [(contains? #{{\"{task_filter}\"}} ?marker)]
+            ]
+        """
```

Comment:
I had some trouble with this, using the `task` query style doesn't work with the `logseq.DB.datascriptQuery` API.  The contains/marker approach is a common alternative used is lots of Logseq query examples.

I'm no expert on clojure and datascript, but I suspect this is because `task` is a clojure function where as the API requires just the expanded datascript query.